tests/package_tests/k8s_test.py

```diff
@@ -0,0 +1,207 @@
+        )
+    except subprocess.CalledProcessError:
+        pass
+    try:
+        subprocess.check_call(
+            "kubectl delete configmap scalyr-config", shell=True,
+        )
+    except subprocess.CalledProcessError:
+        pass
+    try:
+        subprocess.check_call(
+            ["kubectl", "delete", "-f", str(_SCALYR_SERVICE_ACCOUNT_MANIFEST_PATH)],
+        )
+    except subprocess.CalledProcessError:
+        pass
+
+
+def _test(scalyr_api_key: str):
+    # Create agent's service account.
+    subprocess.check_call(
+        ["kubectl", "create", "-f", str(_SCALYR_SERVICE_ACCOUNT_MANIFEST_PATH)]
+    )
+
+    # Define API key
+    subprocess.check_call(
+        ["kubectl", "create", "secret", "generic", "scalyr-api-key", f"--from-literal=scalyr-api-key={scalyr_api_key}"]
+    )
+
+    # Create configmap
+    subprocess.check_call(
+        [
+            "kubectl", "create", "configmap", "scalyr-config",
+            "--from-literal=SCALYR_K8S_CLUSTER_NAME=ci-agent-k8s-",
+        ]
+    )
+
+    # Modify the manifest for the agent's daemonset.
+    scalyr_agent_manifest_source_path = __SOURCE_ROOT__ / "k8s/scalyr-agent-2.yaml"
+    scalyr_agent_manifest = scalyr_agent_manifest_source_path.read_text()
+
+    # Change the production image name to the local one.
+    scalyr_agent_manifest = re.sub(
+        r"image: scalyr/scalyr-k8s-agent:\d+\.\d+\.\d+",
+        "image: scalyr/scalyr-k8s-agent:k8s_test",
+        scalyr_agent_manifest
+    )
+
+    # Change image pull policy to be able to bull the local image.
+    scalyr_agent_manifest = re.sub(
+        r"imagePullPolicy: \w+", "imagePullPolicy: Never", scalyr_agent_manifest
+    )
+
+    # Create new manifest file for the agent daemonset.
+    tmp_dir = tempfile.TemporaryDirectory(prefix="scalyr-agent-k8s-test")
+
+    scalyr_agent_manifest_path = pl.Path(tmp_dir.name) / "scalyr-agent-2.yaml"
+
+    scalyr_agent_manifest_path.write_text(scalyr_agent_manifest)
+
+    # Create agent's daemonset.
+    subprocess.check_call([
+        "kubectl", "create", "-f", str(scalyr_agent_manifest_path)
+    ])
+
+    # Get name of the created pod.
+    pod_name = subprocess.check_output(
+        "kubectl get pods --sort-by=.metadata.creationTimestamp -o jsonpath=\"{.items[-1].metadata.name}\"", shell=True
+    ).decode().strip()
+
+    # Wait a little.
+    time.sleep(3)
+
+    # Execute tail -f command on the agent.log inside the pod to read its content.
+    agent_log_tail_process = subprocess.Popen(
+        ["kubectl", "exec", pod_name, "--container", "scalyr-agent", "--", "tail", "-f", "-n+1",
+         "/var/log/scalyr-agent-2/agent.log"],
+        stdout=subprocess.PIPE
+    )
+    found_errors = []
+
+    # Read lines from agent.log.
+    pipe_reader = PipeReader(pipe=agent_log_tail_process.stdout)
+
+    timeout_time = datetime.datetime.now() + datetime.timedelta(seconds=COMMON_TIMEOUT)
+    error_lines = []
+    try:
+        while True:
```

Comment:
Same here, we probably also need a timeout here so we don't end up in infinite ``while True`` loop in case of an error or similar.